cmd/snap/cmd_run_test.go

```diff
@@ -499,3 +501,70 @@ func (s *SnapSuite) TestSnapRunAppIntegrationFromCore(c *check.C) {
 		"snapname.app", "--arg1", "arg2"})
 	c.Check(execEnv, testutil.Contains, "SNAP_REVISION=x2")
 }
+
+func (s *SnapSuite) TestSnapRunXauthorityMigration(c *check.C) {
+	// mock installed snap; happily this also gives us a directory
+	// below /tmp which the Xauthority migration expects.
+	dirs.SetRootDir(c.MkDir())
+	defer func() { dirs.SetRootDir("/") }()
+	defer mockSnapConfine()()
+
+	u, err := user.Current()
+	c.Assert(err, check.IsNil)
+
+	// Ensure XDG_RUNTIME_DIR exists for the user we're testing with
+	os.MkdirAll(filepath.Join(dirs.XdgRuntimeDirBase, u.Uid), 0700)
```

Comment:
We probably want `err = os.MkdirAll(...); c.Assert(err,IsNil)` here. Just for good measure :)